lib/internal/bootstrap_node.js

```diff
@@ -559,9 +563,30 @@
     nativeModule.cache();
     nativeModule.compile();
 
+    if (config.experimentalModules) {
+      NativeModule._nsCache[id] = {
+        default: nativeModule.exports,
+      };
+
+      const descriptors =
+        Object.getOwnPropertyDescriptors(nativeModule.exports);
+      for (const [name, d] of Object.entries(descriptors)) {
+        if (d.set !== undefined || d.get !== undefined || !d.enumerable)
```

Comment:
Why are non-enumerable properties not allowed? Are there examples of such properties in Node.js right now?

Reply:
are there any cases of non enumerable exports being public api?

Reply:
No, apparently there aren't; but there *are* cases where private APIs are exposed as non-enumerable (e.g. `require('events').prototype`). The check LGTM.